src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs

```diff
@@ -0,0 +1,282 @@
+        /// <param name="reader">FileReader</param>
+        /// <param name="connectionProvider">Connection provider</param>
+        public PolicyExecuterBase(IJsonSerializer serializer, IFileReader reader, IConnectionProvider connectionProvider)
+        {
+            this.Serializer = serializer;
+            this.reader = reader;
+            this.connectionProvider = connectionProvider;
+        }
+
+        /// <summary>
+        /// Gets serializer
+        /// </summary>
+        protected IJsonSerializer Serializer { get; }
+
+        /// <summary>
+        /// Executer
+        /// </summary>
+        /// <param name="arguments">Arguments</param>
+        public void Execute(ExecuterSettings arguments)
+        {
+            var content = this.reader.GetFileContent(arguments.Input);
+            var policyDefinition = this.Serializer.Deserialize<PolicyDefinition>(content);
+
+            using (var connection = this.connectionProvider.GetConnection(arguments))
+            using (var projectClient = connection.GetClient<ProjectHttpClient>())
+            using (var policyClient = connection.GetClient<PolicyHttpClient>())
+            using (var gitRepositoryClient = connection.GetClient<GitHttpClient>())
+            {
+                var projects = projectClient.GetProjects();
+
+                foreach (var project in projects.Result)
+                {
+                    if (!policyDefinition.IsProjectAllowed(project))
+                    {
+                        Log.Debug($"Skipping project \"{project.Name}\", because it's not listed in the allowed projects list");
+                        continue;
+                    }
+
+                    Log.Info($"Starting project: {project.Name}");
+
+                    var types = policyClient.GetPolicyTypesAsync(project.Id).Result;
+
+                    var policyConfigurations = policyClient.GetPolicyConfigurationsAsync(project.Id);
+                    var grouppedRepositories =
+                        policyConfigurations.Result
+                            .Where(x => !policyDefinition.IgnoreType(x.Type.Id, types))
+                            .GroupBy(x => x.GetRepositoryId());
+
+                    var repositories = gitRepositoryClient.GetRepositoriesAsync(project.Id);
+
+                    foreach (var repository in repositories.Result)
+                    {
+                        if (!policyDefinition.IsRepositoryAllowed(repository))
+                        {
+                            Log.Debug($"Skipping repository \"{repository.Name}\", because it's not listed in the allowed repositories list");
+                            continue;
+                        }
+
+                        var relevantPolicies = this.GetPoliciesForRepository(policyDefinition.Policies, project, repository);
+                        var tfsPolicies = grouppedRepositories.FirstOrDefault(x => x.Key == repository.Id.ToString());
```

Comment:
`Use AzureDevOps instead of TFS, as on-prem product is now also called Azure DevOps Server. Or rename to just `policies`
```suggestion
                        var policies = grouppedRepositories.FirstOrDefault(x => x.Key == repository.Id.ToString());
```